language/stdlib/modules/RecoveryAddress.move

```diff
@@ -56,14 +59,29 @@ module RecoveryAddress {
         let caps = &borrow_global<RecoveryAddress>(recovery_address).rotation_caps;
         let i = 0;
         let len = Vector::length(caps);
-        while (i < len) {
+        while ({
+            spec {
+                assert i <= len + 1;
+                assert forall j in 0..i: caps[j].account_address != to_recover;
+            };
+            (i <= len)
+        })
+        // (i < len)
+        {
+            // spec {
+            //     assert forall j in 0..i: caps[j].account_address != to_recover;
+            // };
             let cap = Vector::borrow(caps, i);
             if (LibraAccount::key_rotation_capability_address(cap) == &to_recover) {
                 LibraAccount::rotate_authentication_key(cap, new_key);
                 return
             };
             i = i + 1
         };
```

Comment:
Why is the spec block commented out? What means the `// (i < len)`? Please add a TODO if this is draft. Imagine someone wants to read this code and does not care much about spec.

We need to have a convention to have those loop invariant stuff. It makes code pretty unreadable as of now. @shazqadeer

Reply:
Oops I forgot to delete those lines. 
And yes I agree we should rethink how we write loop invariants. Right now it's hard to find the actual termination condition among all the spec code.